### ISSUE TITLE
torch.cat error with empty boxes

### ISSUE THREAD

**Issue by PaulLerner** [timesler/facenet-pytorch#176]
Hi, I have an error when no boxes are detected for the whole batch, even in the first stage (is that expected? I work with images where there may be no faces)

It is caused by `facenet_pytorch/models/utils/detect_face.py:83` : `boxes = torch.cat(boxes, dim=0)`

If it is expected, I guess a simple fix would be to test `if len(boxes) > 0` like in the other stages

Full traceback is https://gist.github.com/PaulLerner/9b3b6feb95065d8395b4db1dca76241b

**Comment by PaulLerner** [comment]
It happens with this image which has an unusual aspect ratio:
![image](https://upload.wikimedia.org/wikipedia/commons/thumb/3/3e/Complete_Codex_Borbonicus.jpg/800px-Complete_Codex_Borbonicus.jpg)

**Comment by PaulLerner** [comment]
OK now I got it:
it’s because the image height is so small (smaller than `min_face_size`?) then in here https://github.com/timesler/facenet-pytorch/blob/master/models/utils/detect_face.py#L57 the `scales` list is empty, thus the `boxes` is also empty

**Comment by pranavs99** [comment]
thank you @PaulLerner, you saved me a lot of time by pasting the solution in this thread.
